scripts/lib/downloadArtifacts.ts

```diff
@@ -0,0 +1,38 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2023.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { $ } from "zx";
+import { pathExists, getRoot } from "../lib/fs";
+import { mkdirp } from "mkdirp";
+
+export function getArtifactID(url: string) {
+  return url.replace(/.*\//, "");
+}
+
+export async function downloadCIArtifact(
+  pkgName: string,
+  artifactUrl: string,
+  destination: string,
+) {
+  const artifactId = getArtifactID(artifactUrl);
+
+  if (!(await pathExists(destination))) {
+    await mkdirp(destination);
+  }
+
+  await $`gh api \
```

Comment:
I'm not sure this is actually worth adding the code for it. We already say in the prereqs it's required. But FYI in some situations you'd want to proactively check that `gh` is installed and give a good error message to the user if it's not which instructs them to install it.

Not necessary for other binaries like `unzip` since unix users should have it already.